### PR TITLE
Add Documentation for getSharedState API for 3rd Party Ext

### PR DESCRIPTION
**Pull request by cdhoffmann**
Adds documentation for Extension devs for the new `getSharedState(_, resolution: _)` API
This API was merged here: https://github.com/adobe/aepsdk-core-ios/pull/800

### CODE REVIEW

File: Documentation/EventHub/BuildingExtensions.md [hunk 1]
```diff
@@ -249,6 +249,21 @@ All extensions are provided a default API to read shared state from another exte
...
+/// - Parameters:
+///   - extensionName: An extension name whose `SharedState` will be returned
+///   - event: If not nil, will retrieve the `SharedState` that corresponds with the event's version, if nil will return the latest `SharedState`
+///   - barrier: If true, the `EventHub` will only return `.set` if `extensionName` has moved past `event`
+///   - resolution: The `SharedStateResolution` to resolve for. E.g: `.lastSet` will resolve for the last set `SharedState`
+/// - Returns: A `SharedStateResult?` for the requested `extensionName` and `event`
+func getSharedState(extensionName: String, event: Event?, barrier: Bool = false, resolution: SharedStateResolution = .any) -> SharedStateResult?
+```
+
+The resolution is used to fetch a specific type of shared state. Using `.any` will just fetch the last shared state, but using `.lastSet`, will fetch the last shared state with a `.set` status. This is useful if you would like to read the cached config before the remote config has been downloaded.
```
Comment (kevinlind) [inline Documentation/EventHub/BuildingExtensions.md:265]: Suggest rewording: "Using `.any` will fetch the last shared state with any status, while using `.lastSet` will fetch the last shared state with status of `.set`."

File: Documentation/EventHub/BuildingExtensions.md [hunk 2]
```diff
@@ -299,6 +314,21 @@ All extensions are provided a default API to read XDM shared state from another
...
+/// - Parameters:
+///   - extensionName: An extension name whose `SharedState` will be returned
+///   - event: If not nil, will retrieve the `SharedState` that corresponds with the event's version, if nil will return the latest `SharedState`
+///   - barrier: If true, the `EventHub` will only return `.set` if `extensionName` has moved past `event`
+///   - resolution: The `SharedStateResolution` to resolve for
+/// - Returns: A `SharedStateResult?` for the requested `extensionName` and `event`
+func getXDMSharedState(extensionName: String, event: Event?, barrier: Bool = false, resolution: SharedStateResolution = .any) -> SharedStateResult?
+```
+
+The resolution is used to fetch a specific type of shared state. Using `.any` will just fetch the last shared state, but using `.lastSet`, will fetch the last shared state with a `.set` status. This is useful if you would like to read the cached config before the remote config has been downloaded.
```
Comment (kevinlind) [inline Documentation/EventHub/BuildingExtensions.md:330]: Suggest rewording: "Using `.any` will fetch the last shared state with any status, while using `.lastSet` will fetch the last shared state with status of `.set`."

File: Documentation/EventHub/BuildingExtensions.md (hunk 1 above)
Comment (cdhoffmann) [reply Documentation/EventHub/BuildingExtensions.md]: Good suggestion. Done 👍

**Review by kevinlind — APPROVED** [review]
